### PR TITLE
Restore Kotlin DSL subproject editing in IDEA

### PR DESCRIPTION
**Pull request by eskatos**
by using vanilla kotlin-compiler-embeddable as `compileOnlyApi`
and patched one as `runtimeOnly`

also fixing dependencies declaration of the patched compiler to match published metadata

### CODE REVIEW

File: subprojects/kotlin-dsl/build.gradle.kts [hunk 1]
```diff
@@ -49,6 +54,12 @@ dependencies {
     implementation(libs.guava)
     implementation(libs.inject)
 
+    implementation(libs.futureKotlin("script-runtime")) {
+        isTransitive = false
+    }
+    implementation(libs.futureKotlin("daemon-embeddable")) {
+        isTransitive = false
+    }
```
Comment (jjohannes) [inline subprojects/kotlin-dsl/build.gradle.kts:62]: Why the two `isTransitive = false`? I t does not look right conceptually and I don't think it has any effect as both things are fully dependencies further down (in `kotlin-compiler-embeddable/build.gradle.kts `).

**Review by jjohannes — CHANGES_REQUESTED** [review]
Thanks for fixing this. Please close the corresponding issue once this is merged: #13360

I have a question about the details of the dependency declarations.

**Comment by eskatos** [comment]
@bot-gradle test RFN

File: subprojects/kotlin-dsl/build.gradle.kts (hunk 1 above)
Comment (eskatos) [reply subprojects/kotlin-dsl/build.gradle.kts]: You're right, it doesn't make sense, fixed, thanks!

**Review by jjohannes — APPROVED** [review]

**Pull request merged**